src/SFA.DAS.AssessorService.Application/Handlers/EpaOrganisationHandlers/CreateEpaOrganisationStandardHandler.cs

```diff
@@ -59,7 +59,14 @@ public async Task<string> Handle(CreateEpaOrganisationStandardRequest request, C
 
             var organisationStandard = MapOrganisationStandardRequestToOrganisationStandard(request);
            
-            return await _registerRepository.CreateEpaOrganisationStandard(organisationStandard, request.DeliveryAreas);
+            if (request.ApplyFollowingWithdrawal)
```

Comment:
So if it's an apply following a withdrawal, we update the records which is great. If it's not then we add the records which is also great.

One scenario I thought of which i'm unsure if it's a valid scenario is, can the application following a withdrawal be for many versions? If it can, then you could be in a scenario where you withdrew 1.0 but never 1.1 as that's what you're applying for now.

In that scenario you need to update 1.0 and add 1.1.

If you withdrew from a version, and still hold the standard, then you can only opt in / short apply for one version, so the above scenario is invalid in this case.

If you've withdrawn from the entire standard, then choose to apply again, can you choose to apply for more than one version? in which case the current logic would fail.

Reply:
The behaviour is that :
1. System will detect that it has previously been withdrawn
2. System will update the versions but WILL NOT create the entries for the new additional version selected

Reply:
Modified SQL in repository- it will check whether the standard version(s) passed into the query exists - if it does not then it will create it